src/query/interpreter.cpp

```diff
@@ -3260,6 +3406,14 @@ void Interpreter::Commit() {
   utils::OnScopeExit clean_status(
       [this]() { transaction_status_.store(TransactionStatus::IDLE, std::memory_order_release); });
 
+  auto current_storage_mode = interpreter_context_->db->GetStorageMode();
+  auto creation_mode = db_accessor_->GetCreationStorageMode();
+  if (creation_mode != storage::StorageMode::ON_DISK_TRANSACTIONAL &&
+      current_storage_mode == storage::StorageMode::ON_DISK_TRANSACTIONAL) {
+    throw QueryException(
```

Comment:
Switching to on-disk during a transaction should be impossible.

Reply:
but I have a check:
```cpp
if (in_explicit_transaction) {
    throw StorageModeModificationInMulticommandTxException();
}
```

Reply:
The `void Interpreter::Commit()` is called on COMMIT, right?
The `STORAGE MODE ON_DISK_TRANSACTIONAL;` will fail because we are in a transaction. 
So when is this check in Commit() true?

Reply:
which check?

Reply:
```
  if (creation_mode != storage::StorageMode::ON_DISK_TRANSACTIONAL &&
      current_storage_mode == storage::StorageMode::ON_DISK_TRANSACTIONAL) {
```
That check is done in the Commit() and should never be true, right?
Even if it could be true in the current implementation, I think we should make changes to the code so that it never is true.
We don't allow mode changes in a transaction and we do not allow it if other users are actively using the database.